### DISCUSSION TITLE
channels, program listings, and favorites gone :-(

### DISCUSSION THREAD

**Discussion by DavidLaidlaw** [OpenSageTV/Community#69] [Q&A]
Sometime in the last 2 weeks, the channels, program listings, and favorites have disappeared from my SageTV setup.  I have a Wiz.bak file dated 8/11, which was the last day something recorded correctly.  It's about 85Mb.  There is also a wiz.bin.tmp file that is about 14Mb.  I've tried copying the Wiz.bak file to Wiz.bin and restarting.  After doing that there was a log message suggesting that it is getting a java EOF error reading the database file.

I'm not sure how to troubleshoot here and would welcome any help.  I am happy to try to provide any additional info that might be helpful.

Thanks!

**Comment by Ugadata** [comment]
Do you have any backups of your setup?  Either backups of SageTV or even entire system (image) backups ?  May come in handy.

What type of tuner(s) are you using?  Over-The-Air, cable card, HD-Pvr/Colossus, etc.

Is this a Windows system, Unraid, or Linux?  

Do you run your Sage sever as an application or Service?
    (If you use the service and you changed the default UserName, sometimes the User name the service 
   uses can revert to the default UserName. )

If you look under tuner setup, do you still see tuner(s) setup in Sage?

Did you make a copy of your Wiz.bin file before trying to replace it?

**Reply by DavidLaidlaw** [reply]
Many thanks for the response and the questions!  Let me try to answer them as much as I can.

I do not have SageTV folder backups, sadly.  I thought I had at least saved one when we migrated to a new machine a year or two ago, but I can't find it (I'm still looking).

I have 3 OTA tuner cards.  They are all Hauppauge 885 PCI cards.

SageTV is running on Windows 10

SageTV runs as an application without the service running.

The System Information shows all three tuners set up.

When I started troubleshooting there was no Wiz.bin file.  There was a Wiz.bak file and a Wiz.bin.tmp file.  The Wiz.bin.tmp file seems to be recreated each time SageTV is started.  The Wiz.bak file is from 8/11, likely near the time the problem started.

**Comment by Ugadata** [comment]
Ok, So if you haven't already made a copy of your Sage folder ....... (probably goes without saying but be sure Sage is shutdown before making a copy.)

From what I am reading it appears something happened to the Wiz.bin file.  Sage will recreate the Wiz.bin, but you will lose any watched statuses and probably favorites as well (I suspect you may already be aware of this).  

Have you tried going into the Setup Video Sources and trying to preview a channel?  If you have tuners but no channels for the tuners - try re-doing the tuner setup. 
It occurs to me that if you have tuners you may need to redo the EPG source.

**Comment by DavidLaidlaw** [comment]
Yeah, I think we're on the same page.

I stopped SageTV, removed the Wiz.bak and Wiz.bin files (copying them into a subdir first), and restarted SageTV.  It came back up moderately functional.  The lineup and listings looked mostly right now, and I was able to touch them up and get them back to where I think they had been.

Recordings all seem to be there, but I guess you are saying that the "watched" indicator isn't recovered from the recording and metadata files.  Those statuses do seem to be missing.

I did lose Favorites, but have recreated those that I can remember.  I suspect I missed some, but we'll probably figure them out going forward.

After getting thing back to this state, I made another copy of all the *.properties *.bin *.bak files in another subdir as a backup.  And I made a calendar notation to do this routinely to avoid the same problem in the future.  I didn't copy the whole SageTV dir because it is almost 1G in size.  What I did copy was only 52M.  Some of the dirs (STVs, userdata, fontcache, jetty) are big to have extra copies of, I think, but let me know if it's risky not to copy everything.

Thanks for your help!

**Comment by Ugadata** [comment]
Yeah, it is good to read things are functioning --- not so good that some stuff probably isn't retrievable.
If you use any plugins you may want to go throughs the installed plugins and see if any are needing updates or in distress (they are probably fine but just in case)

Keep in mind if you do find an old Wiz.bin, you can replace what you have now.  But the longer it goes, the more likely you'll not want to replace the wiz.bin since they can't be merged.

Good Luck
